mlrun/api/api/endpoints/model_endpoints.py

```diff
@@ -0,0 +1,346 @@
+import hashlib
+import json
+from dataclasses import dataclass
+from typing import List, Optional, Dict, Any
+
+import pandas as pd
+from fastapi import APIRouter, Query
+from v3io.dataplane import RaiseForStatus
+
+from mlrun.api import schemas
+from mlrun.config import config
+from mlrun.errors import (
+    MLRunNotFoundError,
+    MLRunInvalidArgumentError,
+)
+from mlrun.utils import logger
+from mlrun.utils.v3io_clients import get_v3io_client, get_frames_client
+
+ENDPOINTS = "monitoring/endpoints"
+ENDPOINT_EVENTS = "monitoring/endpoint_events"
+
+
+@dataclass
+class TimeMetric:
+    tsdb_column: str
+    metric_name: str
+    headers: List[str]
+
+    def to_metric(self, data: pd.DataFrame) -> Optional[schemas.Metric]:
+        if data.empty or self.tsdb_column not in data.columns:
+            return None
+
+        values = data[self.tsdb_column].reset_index().to_numpy()
+        describe = data[self.tsdb_column].describe().to_dict()
+
+        return schemas.Metric(
+            name=self.metric_name,
+            start_timestamp=str(data.index[0]),
+            end_timestamp=str(data.index[-1]),
+            headers=self.headers,
+            values=[(str(timestamp), float(value)) for timestamp, value in values],
+            min=describe["min"],
+            avg=describe["mean"],
+            max=describe["max"],
+        )
+
+    @staticmethod
+    def from_string(name) -> "TimeMetric":
```

Comment:
python 3.7 does not support referencing self class in class. This is something that was added in 3.8. So if you do construction methods instead of factories + typing, you have to substitute a regular type hint with a string type hint.

Reply:
I was aware of the problem, but didn't know this solution, cool 👌